### DISCUSSION TITLE
SSO with Azure AD

### DISCUSSION THREAD

**Discussion by goffity** [directus/directus#10511] [Archive]
I tried setting up Azure AD authentication according to the documentation https://docs.directus.io/configuration/config-options/#oauth-2-0.

But there is a problem with the response coming from Azure AD.

When "Log In with Microsoft" is pressed, a response from Azure AD is redirected to "${doamin}/login?reason=INVALID_USER"

I tried to find a solution from the discussions board but I don't find it.
#7653

This is the log from Directus.
```
${app} | 07:36:46 ✨ request completed GET 302 /auth/login/microsoft?redirect=${PUBLIC_URL}/admin/login?continue 20ms
${app} | 07:36:48 ✨ request completed GET 302 /auth/login/microsoft/callback?code=XXXXXX&state=xxxxxxxx&session_state=xxxxxx 1.1s
${app} | 07:36:48 ✨ request completed GET 304 /admin/login?reason=INVALID_USER 4ms
```

The values ​​I set in .env
```
AUTH_PROVIDERS="microsoft"
AUTH_MICROSOFT_DRIVER="openid"
AUTH_MICROSOFT_CLIENT_ID="XXXXX"
AUTH_MICROSOFT_CLIENT_SECRET="YYYYYY"
AUTH_MICROSOFT_SCOPE="openid profile email"
AUTH_MICROSOFT_ICON="microsoft"
AUTH_MICROSOFT_AUTHORIZE_URL="https://login.microsoftonline.com/zzzzzzzzzzz/oauth2/v2.0/authorize"
AUTH_MICROSOFT_ACCESS_URL="https://login.microsoftonline.com/zzzzzzzzzzz/oauth2/v2.0/token"
```

What version of Directus are you using?
9.2.2

What version of Node.js are you using?
v16

What database are you using?
Postgres 14

What browser are you using?
Chrome

What operating system are you using?
CentOS 7

How are you deploying Directus?
Docker

**Comment by superhp** [comment]
The following setup works for me.

AUTH_PROVIDERS="microsoft"
AUTH_MICROSOFT_DRIVER="openid"
AUTH_MICROSOFT_CLIENT_ID="YOUR CLIENT ID"
AUTH_MICROSOFT_CLIENT_SECRET="YOUR CLIENT SECRET"
AUTH_MICROSOFT_SCOPE="openid email"
AUTH_MICROSOFT_IDENTIFIER_KEY="email"
AUTH_MICROSOFT_ISSUER_URL="https://login.microsoftonline.com/YOUR_TENANT_ID/v2.0/.well-known/openid-configuration"

And then you have to set Microsoft as authentication provider and user's AzureAD email for every user in Directus that want to authenticate with Microsoft/AzureAD.

![image](https://user-images.githubusercontent.com/7535630/146232787-ced6f011-e661-4d23-a2f4-87d5619c2ee3.png)

**Reply by goffity** [reply]
Thanks, @superhp . It's work. I pass ISSUER_URL to .env  file.

**Reply by u12206050** [reply]
Hey, is this still working for you, I have took your exact setup and replaced `CLIENT_ID`,`CLIENT_SECRET` and `TENANT_ID` but not working.

**Reply by PasitT40** [reply]
> Hey, is this still working for you, I have took your exact setup and replaced `CLIENT_ID`,`CLIENT_SECRET` and `TENANT_ID` but not working.

same issue here . you can fixed ?

**Reply by u12206050** [reply]
Yeh I got it working in the end. This is my full env config:


```
AUTH_PROVIDERS=microsoft
AUTH_MICROSOFT_DRIVER=openid
AUTH_MICROSOFT_ICON=business
AUTH_MICROSOFT_CLIENT_ID=744dc8a0-8944-4014-8c75-5f045306d01e
AUTH_MICROSOFT_CLIENT_SECRET=1fb92ecc$a824%46e7!998c$2450c2922a54
AUTH_MICROSOFT_ISSUER_URL=https://login.microsoftonline.com/744dc8a0-8944-4014-8c75-5f045306d01e/v2.0/.well-known/openid-configuration
AUTH_MICROSOFT_SCOPE=openid user.read mail.read
AUTH_MICROSOFT_IDENTIFIER_KEY=email
AUTH_MICROSOFT_REQUIRE_VERIFIED_EMAIL=false
AUTH_MICROSOFT_ALLOW_PUBLIC_REGISTRATION=true
AUTH_MICROSOFT_DEFAULT_ROLE_ID=5e19ff35-fff6-44b5-8846-bcc70869dc19
```

**Reply by rijkvanzanten** [reply]
@u12206050 You just shared ID/Secrets in a public forum. If these aren't just dummy numbers as an example, please rotate your keys immediately

**Reply by u12206050** [reply]
Dummy keys ;)

**Reply by PasitT40** [reply]
Ty guys finally it worked
